olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.cpp

```diff
@@ -510,16 +511,10 @@ SendOutcome NetworkWinHttp::Send(NetworkRequest request,
 }
 
 void NetworkWinHttp::Cancel(RequestId id) {
-  std::unique_lock<std::recursive_mutex> lock(mutex_);
-  auto it = http_requests_.find(id);
-  if (it == http_requests_.end()) {
-    return;
-  }
-
-  // Just closing the handle cancels the request
-  if (it->second->http_request) {
-    WinHttpCloseHandle(it->second->http_request);
-    it->second->http_request = NULL;
+  auto handle = FindRequest(id);
+  if (handle) {
+    WinHttpCloseHandle(handle->http_request);
+    FreeRequest(handle);
```

Comment:
Looks like there are can be a data race, because this implementation uses locking in different way than previous implementation, which protected the whole method call.